assets/js/src/newsletter_editor/initializer.jsx

```diff
@@ -25,6 +25,14 @@ const initializeEditor = (config) => {
 
   if (!editorContainer || !window.EditorApplication) return;
 
+  if (
+    !parseInt(window.mailpoet_woocommerce_settings.use_mailpoet_editor, 10)
+    && getUrlParam('id') === window.mailpoet_woocommerce_settings.transactional_email_id
+  ) {
+    window.location.href = 'admin.php?page=mailpoet-settings#woocommerce';
+    return;
+  }
```

Comment:
I was wondering why this is not done in PHP since 'id' is a real URL param. Here: https://github.com/mailpoet/mailpoet/blob/master/lib/AdminPages/Pages/NewsletterEditor.php#L39

But I guess it's not too important? I mean, in PHP it's safe and it will occur sooner. You could also use `WPNotice` (although that won't survive redirect I guess but you can still use some `?param=...` and do `WPNotice` in PHP). Does this need to be secure? Is it worth it to move to PHP? It would also be a bit simpler - no new routes and stuff.